### ISSUE TITLE
More features for GET templates/repositories

### ISSUE THREAD

**Issue by tetchel** [eclipse/codewind#179]
As we continue to design the repository management, we need two new fields to be added to each repository object.

1. (easy) We should have a boolean flag for indicating a repository is a Codewind default. We will then block the user from deleting this repository because it is not straightforward to re-add it (it is impossible with appsody which does not have a corresponding url). The user can still disable the repository.

2. We want to be able to show the project styles that a repository has. This would be done by fetching the `index.json` and removing duplicate `projectStyles`. This will be a string array. 

2. is a little more complex because in theory the projectStyles can change at any time when a template is added or removed, but we also don't want to have to call out to fetch and parse `index.json` every time the repository list is requested. So caching should be done, but I'm not sure how we would know when to invalidate that cache.
We could fetch it just once on each Codewind restart, and provide an option (in the querystring?) to force a full refresh, and pass that option whenever the user explicitly clicks Refresh.

**Comment by deboer-tim** [comment]
FWIW, for 1 Appsody is now essentially implemented as a URL callback, so it could be pinged once and stored in the list (which I think it probably should be) and then deleted and added back. I don't love the thought of making it non-removable either way, but maybe once we get the support for enabling/disabling repos that won't matter.

**Comment by tetchel** [comment]
>  I don't love the thought of making it non-removable either way, but maybe once we get the support for enabling/disabling repos that won't matter.

Yeah, that's the idea, it can still be "disabled", but not "deleted", so it's always in the list and the user can always get it back if they want - though its templates won't show up in the creation wizard.

**Comment by tobespc** [comment]
for 1) We now have enabling and disabling repos so do we still want to prevent them removing the defaults ?  That is a valid use case to be able to configure to not have anything we ship 

for 2) At present there are only 2 styles, "codewind" and "appsody".  We can add to the information you get when calling /api/v1/templates/repositories to also include an array of styles.   For example

```
codewind git:(master) ✗ curl http://localhost:9090/api/v1/templates/repositories
[
  {
    "url": "https://raw.githubusercontent.com/kabanero-io/codewind-templates/master/devfiles/index.json",
    "description": "Standard Codewind templates.",
    "enabled": true,
    "styles": ["appsody","codewind"]
  }
]
```

**Comment by tobespc** [comment]
@rwalle61 Can you work on implementing (2) please

**Comment by rwalle61** [comment]
sure

**Comment by jopit** [comment]
Yes, we still want to prevent the user from removing the defaults.

**Comment by rwalle61** [comment]
https://github.com/eclipse/codewind/pull/382 resolved part 1 of this
